src/app/backend/resource/job/detail.go

```diff
@@ -95,3 +99,50 @@ func getJobDetail(job *batch.Job, metricClient metricapi.MetricClient,
 		Completions:     job.Spec.Completions,
 	}
 }
+
+// DeleteJob with given name in given namespace and related pods.
+func DeleteJob(client k8sClient.Interface, namespace, name string) error {
+
+	log.Printf("Deleting %s job from %s namespace", name, namespace)
+
+	job, err := client.BatchV1().Jobs(namespace).Get(name, metaV1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	count := 0
+	*job.Spec.Parallelism = int32(count)
+	_, err = client.BatchV1().Jobs(namespace).Update(job)
```

Comment:
I'm still not sure if this is the right approach. This is non-blocking operation which means that it will trigger the scale operation and proceed with rest of the code. In case cluster is heavily used this could cause same effect as before. Pods will be deleted and job will have time to recreate some of them before it gets scaled and deleted.

Kubectl executed blocking operation which means that it waits until scale operation is done and only then proceeds.